libs/api/domains/powerbi/src/lib/powerbi.resolver.ts

```diff
@@ -0,0 +1,176 @@
+import { Configuration, ConfidentialClientApplication } from '@azure/msal-node'
+import { Inject } from '@nestjs/common'
+import { Args, Query, Resolver } from '@nestjs/graphql'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import { PowerBiEmbedTokenInput } from './dto/powerbiEmbedToken.input'
+import { PowerBiEmbedTokenResponse } from './models/powerbiEmbedTokenResponse'
+
+type Owner = 'Fiskistofa'
+const BASE_URL = 'https://api.powerbi.com/v1.0/myorg'
+
+@Resolver()
+export class PowerBiResolver {
+  constructor(
+    @Inject(LOGGER_PROVIDER)
+    private readonly logger: Logger,
+  ) {}
+
+  private async getAccessToken(owner: Owner) {
+    let clientId = ''
+    let clientSecret = ''
+    let tenantId = ''
+
+    switch (owner) {
+      case 'Fiskistofa':
+        {
+          clientId = process.env.FISKISTOFA_POWERBI_CLIENT_ID as string
+          clientSecret = process.env.FISKISTOFA_POWERBI_CLIENT_SECRET as string
+          tenantId = process.env.FISKISTOFA_POWERBI_TENANT_ID as string
```

Comment:
Should use [ConfigModule](https://github.com/island-is/island.is/tree/main/libs/nest/config) to get runtime config validation.

These variables also need to be defined in the infra file and parameter store for all environments.

Reply:
Done